image/tarexport/save.go

```diff
@@ -268,16 +274,17 @@ func (s *saveSession) save(outStream io.Writer) error {
 			reposLegacy[familiarName][ref.Tag()] = digest.Digest(imageDescr.layers[len(imageDescr.layers)-1]).Encoded()
 			repoTags = append(repoTags, reference.FamiliarString(ref))
 
-			manifestDescriptors = append(manifestDescriptors, ocispec.Descriptor{
-				MediaType: ocispec.MediaTypeImageManifest,
-				Digest:    dgst,
-				Size:      size,
-				Platform:  m.Config.Platform,
-				Annotations: map[string]string{
-					images.AnnotationImageName: ref.String(),
-					ocispec.AnnotationRefName:  ref.Tag(),
-				},
-			})
+			taggedManifest := untaggedMfstDesc
+			taggedManifest.Annotations = map[string]string{
+				images.AnnotationImageName: ref.String(),
+				ocispec.AnnotationRefName:  ref.Tag(),
```

Comment:
Hm.. interesting; so that would mean duplicating the same information (and implicitly; requiring the reader of that information having to parse the format to get the tag only?).

One thing I notice in that section is that it looks like `separator` as part of components seems to have been conflated with `separator` between name and tag (or digest);

> - **org.opencontainers.image.ref.name** Name of the reference for a target (string).
>     - SHOULD only be considered valid when on descriptors on `index.json` within [image layout](image-layout.md).
>     - Character set of the value SHOULD conform to alphanum of `A-Za-z0-9` and separator set of `-._:@/+`
>     - A valid reference matches the following [grammar](considerations.md#ebnf):
> 
>       ```ebnf
>       ref       ::= component ("/" component)*
>       component ::= alphanum (separator alphanum)*
>       alphanum  ::= [A-Za-z0-9]+
>       separator ::= [-._:@+] | "--"
>       ```

The [distribution/reference grammar](https://pkg.go.dev/github.com/distribution/reference) uses a different definition;

```
separator := /[_.]|__|[-]*/
```

- it only allows `-._` as separators
- 2 consecutive underscores (`__`)
- zero or more hypens (`-`)

The  `:`, `@` would be to separate tag or digest, and 


If I read the OCI spec there correctly, something like `docker.io/github.com/he:llo@sha256+fo--b_ar` would be a valid value, but not parseable with the distribution/reference package :grimace:

Reply:
Hm.. for `org.opencontainers.image.base.name`, there's a mention that it should be in the format described in distribution/reference; should that also be done for the above?

> - **org.opencontainers.image.base.name** Image reference of the image this image is based on (string)
>  - This SHOULD be image references in the format defined by [distribution/distribution][distribution-reference].